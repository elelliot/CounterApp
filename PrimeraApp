//import React from 'react'; //imr+tab para importar react , si usas jsx debes importarlo, pero ya con las nuevas versiones no

//import { Fragment } from "react";

import PropTypes from 'prop-types';

// functional components (Componentes)
//props es el objeto propiedades, pero para no poner props.saludo, mejor lo desestructuramos
const PrimeraApp = ( { saludo, subtitulo } ) => { 
    


    /* solo puedes retornar más de un elemento si lo agrupas en uno solo, ejemplo en un div,
     o si no lo necesitas, puedes usar fragments (también puedes crear un fragment dejando vacío los tags y los cierras) */
    return (
        //<div>
        //<Fragment>
        <>
            <h1> { saludo } </h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
            <p> {subtitulo} </p>
        </>
        //</Fragment>
        //</div>
    );

    
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired //La propiedad saludo debe ser string y es obligatoria
}

//mandamos propiedades por defecto, aunque tambien se puede de forma tradicional ejemlo : { saludo, subtitulo= 'Valor' }
PrimeraApp.defaultProps = {
    subtitulo: 'Subtitulo miado'
}

export default PrimeraApp;