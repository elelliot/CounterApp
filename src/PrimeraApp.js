//import React from 'react'; 
//import { Fragment } from "react";

import PropTypes from 'prop-types';
const PrimeraApp = ( { saludo, subtitulo } ) => { 
    
    return (
        //<div>
        //<Fragment>
        <>
            <h1>{ saludo }</h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
            <p>{subtitulo}</p>
        </>
        //</Fragment>
        //</div>
    );

    
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired 
}

PrimeraApp.defaultProps = {
    subtitulo: 'Subtitulo miado'
}

export default PrimeraApp;