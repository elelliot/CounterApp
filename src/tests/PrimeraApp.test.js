//import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme' //shallow es parecido al render
import '@testing-library/jest-dom';

describe('Pruebas en <PrimeraApp />', () => {
    
    //esto usaba el extend-expect
    // test('Debe mostrar el mensaje "Verguerito saludo" ', () => {
        

    //     const saludo = 'Verguerito saludo';

    //     const {getByText} = render( <PrimeraApp saludo= { saludo } />);


    //     expect( getByText( saludo ) ).toBeInTheDocument();
    // })


    test(' Debe mostrar <PrimeraApp /> Correctamente ', () => {
        
        const saludo = 'Saludo Verguero';
        const wrapper = shallow( <PrimeraApp saludo = { saludo } />);

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Saludo Verguero';
        const subTitulo = 'Subtitulo Verguero';
        const wrapper = shallow( 
            <PrimeraApp 
            saludo = { saludo }
            subtitulo = { subTitulo }
             />
             );

        const textoParrafo = wrapper.find('p').text();
        //console.log(textoParrafo);

        expect( textoParrafo).toBe( subTitulo );
    })
    
    
    
})
