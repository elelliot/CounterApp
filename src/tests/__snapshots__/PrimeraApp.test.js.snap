// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp />  Debe mostrar <PrimeraApp /> Correctamente  1`] = `
<Fragment>
  <h1>
    Saludo Verguero
  </h1>
  <p>
    Subtitulo miado
  </p>
</Fragment>
`;
